From 5a87476b9fdb0b1ca66515fde4e9a95e6f88be58 Mon Sep 17 00:00:00 2001
From: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Date: Thu, 3 Feb 2022 18:38:15 +0200
Subject: [PATCH 57/64] ddr: Remove CONFIG_SYS_ERRATUM_ERR050543

Dropped the SYS_ERRATUM_ERR050543 config from KConfig,
due to its redundancy. To check whether or not the Erratum
should be enabled, the status property of the '/ddr' node from
U-Boot's DT is now checked, which is set accordingly by ATF.

Issue: ALB-8353
Upstream-Status: Pending 

Signed-off-by: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/cpu/armv8/s32/fdt.c     |  4 ----
 board/freescale/s32-gen1/Kconfig | 12 ------------
 2 files changed, 16 deletions(-)

diff --git a/arch/arm/cpu/armv8/s32/fdt.c b/arch/arm/cpu/armv8/s32/fdt.c
index 77542d8c5b..0bf411c5d5 100644
--- a/arch/arm/cpu/armv8/s32/fdt.c
+++ b/arch/arm/cpu/armv8/s32/fdt.c
@@ -168,7 +168,6 @@ void ft_fixup_cpu(void *blob)
 }
 #endif /* CONFIG_MP */
 
-#ifdef CONFIG_SYS_ERRATUM_ERR050543
 static void ft_fixup_ddr_polling(const void *old_blob, void *new_blob)
 {
 	int off, ret;
@@ -205,7 +204,6 @@ static void ft_fixup_ddr_polling(const void *old_blob, void *new_blob)
 		printf("WARNING: Could not fix up the Linux DT, err=%s\n",
 		       fdt_strerror(ret));
 }
-#endif
 
 static void hide_sram(bd_t *bd)
 {
@@ -906,9 +904,7 @@ void ft_cpu_setup(void *blob, bd_t *bd)
 #endif
 
 	ft_fixup_memory(blob, bd);
-#ifdef CONFIG_SYS_ERRATUM_ERR050543
 	ft_fixup_ddr_polling(gd->fdt_blob, blob);
-#endif
 #ifdef CONFIG_S32_ATF_BOOT_FLOW
 	ft_fixup_scmi(blob);
 	ft_fixup_atf(gd->fdt_blob, blob);
diff --git a/board/freescale/s32-gen1/Kconfig b/board/freescale/s32-gen1/Kconfig
index 9829abefd9..e9e342001c 100644
--- a/board/freescale/s32-gen1/Kconfig
+++ b/board/freescale/s32-gen1/Kconfig
@@ -73,18 +73,6 @@ config SYS_CONFIG_NAME
 	default "s32r45" if NXP_S32R45
 	default "s32g399a" if NXP_S32G3XX
 
-config SYS_ERRATUM_ERR050543
-	bool "Workaround for NXP Erratum ERR050543"
-	default y if (NXP_S32G2XX || NXP_S32R45) && \
-			!TARGET_TYPE_S32GEN1_EMULATOR
-	help
-	  This option enables a workaround for NXP Erratum ERR050543
-	  (This erratum applies for LPDDR4 memories only. The memories require
-	  a periodic refresh to maintain memory contents and the refresh rate
-	  has to vary over temperature and S32G contains an hardware-based
-	  automatism to carry this dynamic adjustment out. This automatism however
-	  does not work under all conditions.)
-
 # Size of SRAM area used by U-boot during the first boot stage. This includes
 # all of the sections in the System.map, plus the runtime stack which
 # is constructed starting from this offset downwards. Note this has nothing
-- 
2.17.1

